### ISSUE TITLE
show assigned hotkeys on top of action icons

### ISSUE THREAD

**Issue by alexeyOnGitHub** [civfanatics/CQUI_Community-Edition#227]
hey @the-m4a  , do you think there is a way to display hotkeys (letters/numbers/etc) on top of action icons for units?
e.g. show "F", "R", etc on top of these Builder actions:
![image](https://user-images.githubusercontent.com/239741/103142657-c1d93b80-46bb-11eb-8504-07041847cc62.png)

and same for other units. 

I understand not everyone would want to use or see those, so these hints can be either disabled in options, or only shown when a certain key is pressed, like SHIFT, CTRL, ALT, "?", etc.

**Comment by the-m4a** [comment]
hmm.. maybe?  Not sure if it'd require static graphics of the letters themselves, but it's possible.  it'd be some graphic work, I think?  hmm

**Comment by alexeyOnGitHub** [comment]
if we can just show letters and numbers instead of custom graphics, I think that would be perfectly fine. if that is possible, of course

**Comment by alexeyOnGitHub** [comment]
tagging @alimulhuda  here - he may be interested in this feature, I suppose.

also... it may be a long shot, but here is another idea I have in mind that may influence how this is implemented (if it gets implemented) - 
would be awesome to auto-assign hotkeys to additional unit actions, like builders creating ziggurats or whatnot. this is not something that builders have in the base setup, and a builder can acquire a ton of those improvements, or lose them at any moment later in the game.
I would envision those additional actions to be auto-mapped on numbers like ALT+1 , ALT+2, etc. this would make building those improvements en masse easier. (you probably already noticed I tend to prioritize keyboard over mouse as much as it is practical - that is  to avoid carpal tunnel and related injuries). 

the reason why this is related to the action icons shown is that those would not be predetermined actions, and they may be auto-assigned to actions, so it would be definitely useful to see those keys shown next to the actions.

**Comment by alimulhuda** [comment]
I would avoid using graphics for this kind of thing. Keep in mind that the user can remap keys for CQUI added ones by having them in the 'cqui_settings_local.sql' file. The unmodded keys can also be modified by the user and I don't know if there is a way to read which keys are assigned to what action for those ones.

I would lean more towards modifying the text returned as tooltip to include the shortcut key. Something like `button.settooltip(button.gettooltip + "(" + shortcut key string + ")" )`. I just don't know what is required to make that happen and if it's technically possible.

**Comment by alexeyOnGitHub** [comment]
side note on keys mapping in general - it would be best to let users remap them in the game UI without editing files. 
the hotkey values could be loaded from either config files or lua files and then shown in either  tooltips or as letters/numbers on top of icons. I think from usability point of view, I would prefer the hotkey values to be always shown - especially useful for those operations that are dynamic in nature (like tile improvements temporarily granted by City States or "nuke cleanup", etc). I think having this would eliminate half of the mouse clicks required for builders.

**Comment by alimulhuda** [comment]
I agree, ideally we would have the remapping in the UI. But when I was looking into that some time ago, I couldn't find any way to have a mod write to any persistent configuration files. The most I've seen we can do is write it to the savegame, that would mean any remaps would have to be done for every new game.
My personal opinion is that shortcuts should always be shown in the tooltip, I don't consider it too intrusive and other programs like Word, Excel, etc have it always shown so I think most people just subconsciously filter them out unless they are looking for it.
